flocker/control/_diffing.py

```diff
@@ -111,9 +262,23 @@ class Diff(PClass):
     changes = pvector_field(object)
 
     def apply(self, obj):
+        proxy = _EvolverProxy(original=obj)
         for c in self.changes:
-            obj = c.apply(obj)
-        return obj
+            if len(c.path) > 0:
+                proxy = c.apply(proxy)
+            else:
+                assert type(c) is _Set
+                proxy = _EvolverProxy(original=c.value)
+        try:
+            return proxy.commit()
+        except:
+            # Imported here to avoid circular dependencies.
+            from ._persistence import wire_encode
```

Comment:
I wasn't able to figure out the path of the circularity this avoids.  Long-term it might make sense to factor the basic stuff into a separate module so there's no circularity and no nested imports are required.